### ISSUE TITLE
Challenge: Use the Rest Parameter with Function Parameters

### ISSUE THREAD

**Issue by lasjorg** [freeCodeCamp/freeCodeCamp#49152]
### Describe the Issue

The challenge is using the `reduce` method which hasn't been introduced yet. I believe it is an artifact of the current section order (ES6 now comes before functional programming).

You don't really need to understand how `reduce` works to solve the challenge but just to avoid any confusion it might be better to change it to use a simple for loop.

The guide post will need to be updated as well if this change is made.

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/es6/use-the-rest-parameter-with-function-parameters

### Your code

```
const sum = (...args) => {
  let total = 0;
  for (let i = 0; i < args.length; i++) {
    total += args[i];
  }
  return total;
}

console.log(sum(1, 2, 3, 4)); // 10
```

### Expected behavior

Should not contain code that is unfamiliar to the student.

### System

- Device: [e.g. iPhone 6, Laptop]
- OS: [e.g. iOS 14, Windows 10, Ubuntu 20.04]
- Browser: [e.g. Chrome, Safari]
- Version: [e.g. 22]

### Additional context

Forum post: https://forum.freecodecamp.org/t/a-post-in-freecodecamp-challenge-guide-use-the-rest-parameter-with-function-parameters-requires-staff-attention/587761

Guide post: https://forum.freecodecamp.org/t/freecodecamp-challenge-guide-use-the-rest-parameter-with-function-parameters/301221

**Comment by sumitshinde-84** [comment]
hey i am interested in that issue can you assign it to me ?

**Comment by sumitshinde-84** [comment]
hey i have fix this issue let you check PR #49525
